custom_components/apple_tv/__init__.py

```diff
@@ -177,8 +181,19 @@ def connection_closed(self):
 
     async def connect(self):
         """Connect to device."""
-        self._is_on = True
-        self._start_connect_loop()
+        if self._is_pwr_mgmt_on and self._is_already_connected:
+            _LOGGER.debug("Turning the device on")
+            await self.atv.power.turn_on()
+            self._update_state(connected=True)
+        else:
+            self._is_on = True
+            self._start_connect_loop()
+
+    async def softdisconnect(self):
+        """Turn off device."""
+        _LOGGER.debug("Turning the device off")
+        await self.atv.power.turn_off()
+        self._update_state(disconnected=True)
 
     async def disconnect(self):
```

Comment:
I would prefer if we could let the manager deal with everything related to power management. So, I suggest that we rename current `disconnect` to `close_connection` and `softdisconnect` to just `disconnect`. Then we add the logic from `media_player` and `remote` in there:

```python
async def disconnect(self):
    """Disconnect from device."""
    if self._is_pwr_mgmt_on:
        _LOGGER.debug("Turning the device off")
        await self.atv.power.turn_off()
        self._update_state(disconnected=True)
    else:
        await self._close_connection()
```
More or less. We can then revert the changes in said platforms.